lib/index.js

```diff
@@ -5,7 +5,63 @@ const spawn = require('child_process').spawn;
 const spawnSync = require('child_process').spawnSync;
 const JSON = require('./json-buffer');
 const isElectron = require('is-electron');
+const fs = require('fs');
+const os = require('os');
+const index = [0];
+const timeStamp = new Date().toISOString();
+const net = require('net');
+let clientConnected = false;
+let client;
+const READFILETIMELIMIT = 5000;
+const debugMode = false;
 
+const responseFolder = path.join(os.tmpdir(), 'sync_rpc_responses');
+if (!fs.existsSync(responseFolder)) {
+  fs.mkdirSync(responseFolder, {recursive: true});
+}
+
+const responseFile = path.join(responseFolder, `responseFile.txt`);
+
+function timeLimitExceeded(startTime) {
+  return startTime + READFILETIMELIMIT < Date.now();
+}
+
+function responseFileReadOnly() {
+  const mode = fs.statSync(responseFile).mode;
+  return (mode & parseInt('777', 8)).toString(8) == '444';
+}
+
+function responseFileReadyForReading() {
+  return fs.existsSync(responseFile) && responseFileReadOnly();
+}
+
+function readResponseFile() {
+  const startTime = Date.now();
+  while (!responseFileReadyForReading()) {
+    // if (timeLimitExceeded(startTime)) {
```

Comment:
does this need to be uncommented?

Reply:
Keeping the while loop without timeout is similar to `spawnSync`, when the child process doesn't finish, and the parent process doesn't proceed. Also, here the time limit is pending on the network turn-around time. On some really poor connection, it may throw unnecessarily...